### DISCUSSION TITLE
How to get ethers provider in the latest version

### DISCUSSION THREAD

**Discussion by yasoll** [wevm/wagmi#2615] [Q&A] [answered]
In the old version, I could get the ethers provider with the following code
``` typescript
const {data: signer, isLoading: signerLoadig, isError: signerError} = useSigner();
const web3 = new Web3((signer?.provider as any).provider);
```
But now the latest version is updated, I don&#39;t know how to get the provider of ethers
The migration documentation says `useSigner`->`useWalletClient `
and noted`The getSigner method has been renamed to getWalletClient, and also returns viem's WalletClient instead of an Ethers.js Signer`
So now I don't know how to get the provider of ethers，I want to get some help~

**Comment by jxom** [comment, marked as answer]
https://wagmi.sh/react/ethers-adapters

**Comment by yasoll** [comment]
Thanks for your reply, I've been focusing on the Hook and didn't notice the rest of the documentation, I would also recommend adding a link to the solution in the aforementioned reminder if possible
